### PR TITLE
feat(serde_v8): better error output

### PR DESCRIPTION
**Pull request by bartlomieju**
The type that was received is now printed as part of a message.

### CODE REVIEW

**Review by levex — COMMENTED** [review]
This is a sweet addition - helps a lot.

LGTM but I would prefer someone else review it, too so this is not a formal Approve :)

File: serde_v8/de.rs [hunk 1]
```diff
@@ -425,7 +427,7 @@ impl<'de, 'a, 'b, 's, 'x> de::Deserializer<'de>
       })
     } else {
       // TODO: improve error
```
Comment (petamoriken) [inline serde_v8/de.rs:429]: Maybe we can get rid of this TODO comment now.

File: serde_v8/error.rs [hunk 2]
```diff
@@ -9,30 +9,41 @@ pub enum Error {
...
-  #[error("serde_v8 error: invalid type, expected: buffer")]
-  ExpectedBuffer,
-  #[error("serde_v8 error: invalid type, expected: detachable")]
-  ExpectedDetachable,
-  #[error("serde_v8 error: invalid type, expected: external")]
-  ExpectedExternal,
-  #[error("serde_v8 error: invalid type, expected: bigint")]
-  ExpectedBigInt,
+  #[error("serde_v8 error: invalid type; expected: boolean, got: {0}")]
+  ExpectedBoolean(String),
```
Comment (littledivy) [inline serde_v8/error.rs:13]:
```suggestion
  ExpectedBoolean(&'static str),
```

File: serde_v8/error.rs [hunk 3]
```diff
@@ -57,3 +68,90 @@ impl serde::de::Error for Error {
     Error::Message(msg.to_string())
   }
 }
+
+pub(crate) fn value_to_type_str(value: v8::Local<v8::Value>) -> String {
```
Comment (littledivy) [inline serde_v8/error.rs:72]:
```suggestion
pub(crate) fn value_to_type_str(value: v8::Local<v8::Value>) -> &'static str {
```

remove `to_string` in the end.

File: serde_v8/tests/de.rs [hunk 4]
```diff
@@ -408,7 +408,7 @@ detest!(
 );
 
 defail!(defail_struct, MathOp, "123", |e| e
-  == Err(Error::ExpectedObject));
+  == Err(Error::ExpectedObject("Number".to_string())));
```
Comment (littledivy) [inline serde_v8/tests/de.rs:411]:
```suggestion
  == Err(Error::ExpectedObject("Number")));
```

File: serde_v8/error.rs (hunk 2 above)
Comment (littledivy) [reply serde_v8/error.rs:13]: Unless there is a reason this cannot be static

**Review by dsherret — APPROVED** [review]
Nice!

**Pull request merged**